### PR TITLE
Enable checkstyle for all projects

### PR DESCRIPTION
**Pull request by tamirkamara**
1. Enable checkstyle plugin with a single definition file
2. Fix style issues
3. Update docker files and remove mvnw

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -99,6 +102,38 @@
...
+                <executions>
+                    <execution>
+                        <id>validate</id>
+                        <phase>validate</phase>
+                        <goals>
+                            <goal>check</goal>
+                        </goals>
+                    </execution>
+                </executions>
+<!--                This doesn't work for some reason so had to reference the project directly. -->
```
Comment (itye-msft) [inline pom.xml:128]: Is there a user story to solve that?

**Review by itye-msft — APPROVED** [review]
Approved with one question...

File: pom.xml (hunk 1 above)
Comment (tamirkamara) [reply pom.xml:128]: Currently no. I just kept it here for reference...

Comment (tamirkamara) [reply pom.xml]: I removed it now.

**Review by shajidissa — APPROVED** [review]

**Pull request merged**